arkit/server.py

```diff
@@ -0,0 +1,256 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This server will communicate with the AR app with websocket.
+cozmo will try 3 times, it controls by the round number self._round (0-2)
+1. see fireworks, surprise
+2. see fireworks, fail launch
+3. see fireworks, super happy and excited
+"""
+import time
+import websockets
+import asyncio
+import threading
+import asyncio
+import sys
+import PIL.Image
+import base64
+import os
+import io
+import random
+import cozmo
+from cozmo.util import degrees, distance_mm, speed_mmps, Pose
+from cozmo.objects import CustomObject, CustomObjectMarkers, CustomObjectTypes
+from lib.vision_cube import VisionCube
+from lib.config import IP_ADDRESS, PORT
+
+stopFlag = False
+
+USE_VIEWER = False
+USE_LOGGING = False
+DEBUG_MODE = True
+PURPLE = cozmo.lights.Color(rgb=(141, 7, 255))
+BLUE = cozmo.lights.Color(rgb=(6, 6, 255))
+YELLOW= cozmo.lights.Color(rgb=(255, 255, 7))
+GREEN = cozmo.lights.Color(rgb=(0, 255, 0))
+ORANGE = cozmo.lights.Color(rgb=(255, 177, 6))
+PINK = cozmo.lights.Color(rgb=(255, 13, 144))
+
+serverT = threading.Thread()
+
+class TapCube(threading.Thread):
+
+    def __init__(self):
+        threading.Thread.__init__(self)
+        self._round = 0
+        self._cube = 0
+        self._robot = None
+        self._wait_for_tap_flag = False
+        self._did_cozmo_tap_flag = False
+        self._tap_action_finished = False
+        self._animation_triggerd = False
+        self._trigger_message = False
+        self._message = ""
+
+        if USE_LOGGING:
+            cozmo.setup_basic_logging()
+        if USE_VIEWER:
+            cozmo.connect_with_tkviewer(self.run)
+
+        else:
+            try:
+                cozmo.connect(self.run)
+            except cozmo.ConnectionError as e:
+                sys.exit("A connection error occurred: %s" % e)
+
+    async def run(self, coz_conn):
+        """The run method runs once Cozmo is connected."""
+        await self.set_up_cozmo(coz_conn)
+
+        while True:
+            await asyncio.sleep(0)
+        pass
+
+    async def set_up_cozmo(self, coz_conn):
+        asyncio.set_event_loop(coz_conn._loop)
+        self._robot = await coz_conn.wait_for_robot()
+
+        global serverT
+        serverT = threading.Timer(0, self.run_server).start()
+        await self._robot.set_lift_height(0, duration=0.5).wait_for_completed()
+        await self._robot.play_anim("anim_launch_wakeup_02").wait_for_completed()
+        await self._robot.play_anim("anim_hiking_lookaround_01").wait_for_completed()
+        await self._robot.play_anim("anim_hiking_lookaround_03").wait_for_completed()
+        await self._robot.play_anim("anim_hiking_lookaround_02").wait_for_completed()
+        await self._robot.go_to_pose(Pose(0, 0, 0, angle_z=degrees(0)), relative_to_robot=True).wait_for_completed()
+        await self._robot.set_head_angle(cozmo.util.Angle(degrees=0)).wait_for_completed()
```

Comment:
Not a big deal, but there is a helper method `cozmo.util.degrees(0)` intended for this